### ISSUE TITLE
Add ability to turn off the lightwave app | make lightwave pip installable separate app

### ISSUE THREAD

**Issue by superryeti** [MIT-LCP/physionet-build#1780]
Context: On the TRA, a  low priority XSS Vulnerability was discovered  on lightwave setting page and following that, the recommendation was to add the ability to turn off lightwave

1. So for that one suggestion was to make lightwave a separate pip installable app and and add it as an optional dependency.
2. or maybe add an variable in .env to toggle lightwave, and  modify the `project/projectfiles/local.py` and `physionet/settings/base.py` to enable/disable lightwave based on .env

Hence opening the issue to start discussion.

**Comment by superryeti** [comment]
Also it looks like  a good idea to  fix the broken links through the html files when doing this

For example a lot of links to physionet are broken everywhere(404 not found)

https://physionet.org/static/lightwave/doc/client-install.html

**Comment by superryeti** [comment]
@alistairewj @tompollard 

For separating the app from Physionet and making it a installable Django app, I have a good idea on how to approach it. However, I had a confusion on how to refactor this one function.



```

def lightwave_server(request):
    """
    Request LightWAVE data for a published database.
    """
    if request.GET.get('action', '') == 'dblist':
        projects = PublishedProject.objects.filter(
            has_wfdb=True, access_policy=AccessPolicy.OPEN, deprecated_files=False
        ).order_by('title', '-version_order')
        dblist = '\n'.join('{}/{}\t{}'.format(p.slug, p.version, p) for p in projects)
    else:
        dblist = None
    return serve_lightwave(query_string=request.GET.urlencode(),
                           root=PUBLIC_ROOT,
                           dbpath=PUBLIC_DBPATH,
                           dblist=dblist,
                           public=True)
                           
```

This part of the function loads a list of projects with open access policy and queries the database. Since we are trying to separate the LightWAVE app, it makes sense that we probably don't want to have some kind of dependency on the Physionet model PublishedProject. My assumption is that we want to do this, as it would allow someone else to use the LightWAVE app on their own Django projects.

1. If we want to do this, the only way I can think of (so far) is to ask people to override the implementation of this function. I'm not sure if that sounds like a good idea.

2. If we want to go down this path (making the pip-installable LightWAVE app reusable on non-Physionet Django projects), should we also add documentation on how to install/compile the LightWAVE app, including the sandboxed-lightwave component and any necessary setup?(i can find the docs on it, but the links inside it are broken)

<img width="1454" alt="image" src="https://user-images.githubusercontent.com/24412619/210635994-7a1ef226-1f3d-412b-a90c-24467307871f.png">

**Comment by tompollard** [comment]
Tagging @bemoody because he has probably has thoughts on this issue.

**Comment by superryeti** [comment]
Hi @bemoody tagging you to bring this on your plate. :)

**Comment by tompollard** [comment]
> Since we are trying to separate the LightWAVE app, it makes sense that we probably don't want to have some kind of dependency on the Physionet model PublishedProject. 

I  think it is going to be tricky to make this entirely independent of the PhysioNet platform. 

Unless @bemoody has a good suggestion, perhaps best to separate the app out using whatever approach makes sense to you, then raise an new issue pointing out that we should remove the dependency on the PublishedProject object.

**Comment by alistairewj** [comment]
Seems like a good application for the almost finalized API, no? `dblist` can be populated off a call to `.../projects/published/`. At the very least, it would remove the explicit use of `PublishedProject`.

**Comment by tompollard** [comment]
> Seems like a good application for the almost finalized API, no? `dblist` can be populated off a call to `.../projects/published/`. At the very least, it would remove the explicit use of `PublishedProject`.

Yes, good thought! `has_wfdb` and `deprecated_files` are probably good fields for us to add to the API.

**Comment by superryeti** [comment]
started work here https://github.com/MIT-LCP/django-lightwave-app

**Comment by superryeti** [comment]
@bemoody  Could you review the start of the work and share you initial thoughts

https://github.com/MIT-LCP/django-lightwave-app

Summary of what we have done so far

1. copy the lightwave app and add some pypi suggested stuff to make a python package
2. Remove dependency to physionet project by asking developers to add variables, function and decorators on the django project settings
3. Add basic readme instructions


Questions

1.  Do we also want to add information on how to install the lightwave application(LightWAVE server)
This guide has information on how to compile it locally, but the links are outdated

https://physionet.org/lightwave/
![image](https://user-images.githubusercontent.com/24412619/214631634-98614eb4-44fd-44db-8ea3-14a2ca8af5ae.png)

2. I think i need to spend a little more time to get how the  the basic idea of  how `sandboxed-lightwave` or the compiled lightwave works. But i wanted to quickly get your feedback to make sure i was on the right track

**Comment by superryeti** [comment]
I found the lightwave server code https://github.com/bemoody/lightwave/ and instructions on how to compile it on our physionet-build project

https://github.com/bemoody/lightwave/
